File: lab/c/somas/soma_numero_ate_que_zero_seja_digitado.c
#include <stdio.h>

// Versão 1 — com while(true)
float soma_numeros() {
    float soma = 0.0, numero;

    while (1) { // laço infinito
        printf("Digite um número (0 para parar): ");
        scanf("%f", &numero);
        if (numero == 0) break;
        soma += numero;
    }

    return soma;
}

// Versão 2 — comprimida
float soma_numeros_comprimido() {
    float soma = 0.0, numero;

    printf("Digite um número (0 para parar): ");
    scanf("%f", &numero);

    while (numero != 0) {
        soma += numero;
        printf("Digite um número (0 para parar): ");
        scanf("%f", &numero);
    }

    return soma;
}

int main(void) {
    printf("%.2f\n", soma_numeros());
    printf("%.2f\n", soma_numeros_comprimido());
    return 0;
}

File: lab/javascript/somas/soma_numero_ate_que_zero_seja_digitado.js
const prompt = require("prompt-sync")({ sigint: true });

// Versão 1 — com while True
function soma_numeros() {
  let soma = 0;
  while (true) {
    let numero = parseFloat(prompt("Digite um número (0 para parar): "));
    if (numero === 0) break;
    soma += numero;
  }
  return soma;
}

// Versão 2 — comprimida
function soma_numeros_comprimido() {
  let soma = 0;
  let numero = parseFloat(prompt("Digite um número (0 para parar): "));
  while (numero !== 0) {
    soma += numero;
    numero = parseFloat(prompt("Digite um número (0 para parar): "));
  }
  return soma;
}

console.log(soma_numeros());
console.log(soma_numeros_comprimido());
